src/client/csfle/client_encryption.rs

```diff
@@ -32,6 +32,12 @@ pub struct ClientEncryption {
     key_vault: Collection<RawDocumentBuf>,
 }
 
+impl std::fmt::Debug for ClientEncryption {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ClientEncryption").finish()
```

Comment:
Is this for redaction purposes? If so, I think it could be good to follow the existing format for `Credential` (which uses `debug_tuple` and "REDACTED" as the body) as well as introduce a test ensuring it is in fact redacted.

Also, are there any other `Debug` implementations that we might want to redact? Maybe in `Client`?